ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/context/KsqlSecurityContextBinderFactory.java

```diff
@@ -89,10 +88,7 @@ public KsqlSecurityContext provide() {
         Optional.ofNullable(request.getHeader(HttpHeaders.AUTHORIZATION));
 
     if (!securityExtension.getUserContextProvider().isPresent()) {
-      return new KsqlSecurityContext(
-          Optional.ofNullable(principal),
-          defaultServiceContextFactory.create(ksqlConfig, authHeader)
-      );
+      return new KsqlSecurityContext(Optional.ofNullable(principal), serviceContext);
```

Comment:
I can't add a comment in the `dispose` method. But you have to make sure not to close this default `serviceContext` on the `dispose`. The `dispose` is called after a REST request finished.

Btw, could you rename this variable to `defaultServiceContext` or `serverServiceContext`? so we know this is used as the default.

Reply:
Removed the dispose.  Sorry, forgot that from our conversation.

Renamed as well.